### ISSUE TITLE
Functions that use Randomness failing

### ISSUE THREAD

**Issue by Rahat-ch** [movementlabsxyz/movement#365]
**Describe the bug**
Although we can deploy modules with randomness - executing any function that calls randomness fails with the following error: 

```

The `Option` is in an invalid state for the operation attempted. The `Option` is `None` while it should be `Some`.

```

The same code when deployed on Aptos Testnet does work as expected. 

**To Reproduce**
1. Go to https://explorer.movementlabs.xyz/account/0xc29415b849fa1173b986fdd074335b9fcc5b2b795d84163be06656cd903c3c01/modules/run/rock_paper_scissors/play?network=testnet
2. Execute the play function on Rock Paper Scissors Module (argument is a number 0-2)
4. When trying to sign the transaction on your wallet the error will appear 

**Expected behavior**
Onchain randomness should generate a number 0-2 and compare it to your input for a rock paper scissors game

**Additional context**

This is the same code which works on Aptos Testnet: 
https://explorer.aptoslabs.com/account/0xeddd5485f97b5a551a9414963cf3815039d8e593a78a2826238d690555818a2e/modules/run/rock_paper_scissors/get_game_result?network=testnet

**Comment by YB1010** [comment]
I also encounter this issue. the code is working on aptos testnet, but not working on movement suzuka.

**Heres the failed transaction on movement suzuka testnet:** https://blue.explorer.movementlabs.xyz/txn/0xac96dec9f4aec9c9adebae0f305db08a60b8925567904b4262c83e0d616f2dc7/userTxnOverview?network=testnet

**And here is the succeed tsx on aptos testnet:**
https://explorer.aptoslabs.com/txn/0xfd4fc2bd7262fcc37d089ce679f4bf1b0c1eedb963b6acfd2608762448cd8147/balanceChange?network=testnet 

I searched around, seem like there's similar issue on aptos mainnet, but I not sure whether this issue has been resolved on the mainnet. [#Mainnet Native Randomness](https://github.com/aptos-labs/aptos-developer-discussions/discussions/226)

**Comment by l-monninger** [comment]
I don't believe we have randomness turned on: https://github.com/movementlabsxyz/aptos-core/blob/7d117d7eb8052d4bc580a23094bed4e1250e1113/types/src/on_chain_config/randomness_config.rs#L205

This also raises several questions about whether the SLA is to match testnet with Aptos testnet or Aptos mainnet. Randomness is not audited: https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-41.md

**Comment by l-monninger** [comment]
Also, as an aside, for testing, you can roll your own (CS)PRNG and use the block timestamp or provide some other noisy parameter. Here's an example with Xoshiro alg that we use in our testing: https://github.com/movementlabsxyz/movement/blob/ce25a3ba5c4ea4055e2429abcbc97fad1a62bad3/networks/suzuka/suzuka-client/src/tests/complex-alice/sources/resource_roulette.move#L71

**Comment by YB1010** [comment]
Got it, thanks for the feedback. Does Suzuka testnet have any plans to involve Oracle partners like Pyth or Chainlink to provide VRF services? Self-implementing PRNG can be flawed and susceptible to exploitation if not designed with rigorous security measures.

**Comment by l-monninger** [comment]
> Self-implementing PRNG can be flawed and susceptible to exploitation if not designed with rigorous security measures.
>>  **for testing,** you can roll your own (CS)PRNG

> Does Suzuka testnet have any plans to involve Oracle partners like Pyth or Chainlink to provide VRF services? 

Yes, we have Pyth and we are supposed to have Supra. I believe our [docs](https://docs.movementnetwork.xyz/devs/oracles) are outdated though, @Rahat-ch.

IMO, if you believe you have requirements that necessitate a VRF, you should use a VRF and make sure you understand all of its assumptions as well. AIP-41 should not really be regarded as a substitute.

For example, AIP-41 does not cover MEV/test-and-abort attacks from quorum participants themselves, e.g., probing for favorable noise. A VRF could.

**Comment by YB1010** [comment]
> Yes, we have Pyth and we are supposed to have Supra. I believe our [docs](https://docs.movementnetwork.xyz/devs/oracles) are outdated though, @Rahat-ch.

I did recognized that Movement can use Pyth for price feed, yet Pyth Entrophy hasn't stated Movement Suzuka Testnet is supported. [Pyth Contract Addresses](https://docs.pyth.network/entropy/contract-addresses)
The situation with Supra is similar. Despite the announced partnership, I still cannot find their contract for Movement Testnet in their [doc](https://docs.supra.com/docs/vrf/networks) 

BTW, I try to reach out both project last week regarding to the VRF support, and I haven't receive any feedback yet.

**Comment by l-monninger** [comment]
I think help should be on the way...

**Comment by l-monninger** [comment]
Our docs can be clearer, but this is the Pyth address: https://explorer.movementlabs.xyz/account/0x9357e76fe965c9956a76181ee49f66d51b7f9c3800182a944ed96be86301e49f/modules/code/batch_price_attestation?network=testnet

Supra has not yet deployed.
